tests/rptest/services/cluster.py

```diff
@@ -13,7 +13,7 @@
 from ducktape.mark._mark import Mark
 
 
-def cluster(log_allow_list=None, **kwargs):
+def cluster(log_allow_list=None, check_allowed_error_logs=True, **kwargs):
```

Comment:
> The test still uses the cluster decorator to reserve the correct number
of nodes. However, the decorator previously required that the Redpanda
service be started, indicated by the presence of logs. To circumvent
this requirement, I added an argument to the decorator to allow ignoring
logs.

this seems like a bit much just to test `--version` works. I think in cmake you could use `ctest()` plus a small bash script to run the Redpanda binary and test its output as part of the normal unit test framework.